src/opam.ml

```diff
@@ -21,24 +21,25 @@ let binary = Path.ofString "opam"
 type t = Cmd.spawn
 
 let make () =
-  let open Promise.O in
+  let open Promise.Syntax in
   Cmd.checkSpawn { bin = binary; args = [] } >>| function
   | Error _ -> None
   | Ok cmd -> Some cmd
 
 let parseSwitchList out =
   let lines = String.split_on_char '\n' out in
   let result =
-    Belt.List.keepMap lines (function
-      | "" -> None
-      | s -> Some (Switch.make s))
+    lines
+    |> List.filter_map (function
```

Comment:
This module should also use Core_kernel

Reply:
So is it all right to completely transition to using Core_kernel?
I didn't do that when I made the PR because I didn't want to make too many unnecessary changes.

Reply:
Okay, it can  be done later. I saw other files ported to Core_kernel, so I thought you were doing it here too.